lib/doorkeeper/models/access_grant_mixin.rb

```diff
@@ -24,6 +24,10 @@ module AccessGrantMixin
       validates :token, uniqueness: true
 
       before_validation :generate_token, on: :create
+
+      def uses_pkce?
+        code_challenge.present? && code_challenge_method.present?
```

Comment:
doesn't really matter for us because no one in their right mind would wnat to use `plain`, but FYI method is not a required field:

```
   code_challenge_method
      OPTIONAL, defaults to "plain" if not present in the request.  Code
      verifier transformation method is "S256" or "plain".
```